pkg/relay/ethereum/tx.go

```diff
@@ -278,3 +268,69 @@ func (c *Chain) TxAcknowledgement(opts *bind.TransactOpts, msg *chantypes.MsgAck
 		ProofHeight:     pbToHandlerHeight(msg.ProofHeight),
 	})
 }
+
+func (c *Chain) SendTx(opts *bind.TransactOpts, msg sdk.Msg, noSend bool) (*gethtypes.Transaction, error) {
+	logger := c.GetChainLogger()
+	opts.NoSend = noSend
+	var (
+		tx  *gethtypes.Transaction
+		err error
+	)
+	switch msg := msg.(type) {
+	case *clienttypes.MsgCreateClient:
+		tx, err = c.TxCreateClient(opts, msg)
+	case *clienttypes.MsgUpdateClient:
+		tx, err = c.TxUpdateClient(opts, msg)
+	case *conntypes.MsgConnectionOpenInit:
+		tx, err = c.TxConnectionOpenInit(opts, msg)
+	case *conntypes.MsgConnectionOpenTry:
+		tx, err = c.TxConnectionOpenTry(opts, msg)
+	case *conntypes.MsgConnectionOpenAck:
+		tx, err = c.TxConnectionOpenAck(opts, msg)
+	case *conntypes.MsgConnectionOpenConfirm:
+		tx, err = c.TxConnectionOpenConfirm(opts, msg)
+	case *chantypes.MsgChannelOpenInit:
+		tx, err = c.TxChannelOpenInit(opts, msg)
+	case *chantypes.MsgChannelOpenTry:
+		tx, err = c.TxChannelOpenTry(opts, msg)
+	case *chantypes.MsgChannelOpenAck:
+		tx, err = c.TxChannelOpenAck(opts, msg)
+	case *chantypes.MsgChannelOpenConfirm:
+		tx, err = c.TxChannelOpenConfirm(opts, msg)
+	case *chantypes.MsgRecvPacket:
+		tx, err = c.TxRecvPacket(opts, msg)
+	case *chantypes.MsgAcknowledgement:
+		tx, err = c.TxAcknowledgement(opts, msg)
+	// case *transfertypes.MsgTransfer:
+	// 	err = c.client.transfer(msg)
+	default:
+		logger.Error("failed to send msg", errors.New("illegal msg type"), "msg", msg)
+		panic("illegal msg type")
+	}
+	return tx, err
+}
+
+func (c *Chain) EstimateGas(ctx context.Context, tx *gethtypes.Transaction) (uint64, error) {
```

Comment:
This function (`EstimateGas`) should be moved to `pkg/client` because this does not depend on `Chain`, etc.

Reply:
Move to client package